File: January/Day17/minFlipsMonoIncr.cpp
class Solution {
public:
    int minFlipsMonoIncr(string s) {
      int countFlip = 0;
      int countOne = 0;
      for(int i = 0 ;i<s.size();i++)
      {
          if(s[i] == '1')
          {
             countOne++; 
          }
          if(s[i] == '0')
          {
              if(countOne>0)
              {
                  countFlip++;
              }
          }
      if(countOne < countFlip)
      {
          countFlip = countOne;
      }
      }
      return countFlip;
    }
};

File: January/Day17/minFlipsMonoIncr.java
class Solution {
    public int minFlipsMonoIncr(String s) {
        int countone=0;
        int flip=0;
        for(char c:s.toCharArray()){
            if(c=='1')countone++;
            if(c=='0' && countone>0)flip++;
           flip= Math.min(countone,flip);
        }
        return flip;
    }
}
